crates/bgp-pkt/src/wire/serializer/path_attribute/bgp_sid.rs

```diff
@@ -113,3 +134,108 @@ impl WritablePdu<SrgbWritingError> for SegmentRoutingGlobalBlock {
         Ok(())
     }
 }
+
+#[derive(WritingError, Eq, PartialEq, Clone, Debug)]
+pub enum SRv6ServiceSubTlvWritingError {
+    StdIOError(#[from_std_io_error] String),
+    SRv6ServiceSubSubTlvError(#[from] SRv6ServiceSubSubTlvWritingError),
+}
+
+impl WritablePdu<SRv6ServiceSubTlvWritingError> for SRv6ServiceSubTlv {
+    const BASE_LENGTH: usize = 3;
+
+    fn len(&self) -> usize {
+        Self::BASE_LENGTH
+            + match self {
+                SRv6ServiceSubTlv::SRv6SIDInformation { subsubtlvs, .. } => {
+                    1 + 16
+                        + 1
+                        + 2
+                        + 1
+                        + subsubtlvs
+                            .iter()
+                            .map(|subsubtlv| subsubtlv.len())
+                            .sum::<usize>()
+                }
+                SRv6ServiceSubTlv::Unknown { value, .. } => value.len(),
+            }
+    }
+
+    fn write<T: Write>(&self, writer: &mut T) -> Result<(), SRv6ServiceSubTlvWritingError>
+    where
+        Self: Sized,
```

Comment:
why this `where Self: Sized` is included everywhere? Is it needed? I think the code should compile fine without it

Reply:
compiles fine without it, i think it's Rustrover that adds it automatically when I impl a trait. i'll remove it

Reply:
I think it adds it because you have it in the trait definition
```rs

/// Generic trait for Readable Protocol Data Unit that doesn't need any external
/// input while parsing the packet.
pub trait ReadablePdu<'a, Error: Debug> {
    fn from_wire(buf: Span<'a>) -> IResult<Span<'a>, Self, Error>
    where
        Self: Sized;
}
```

Reply:
Yes it does the same for me, and I removed them manually. I've no idea why it does that.

Reply:
It probably just copies the trait bounds from the trait definition to avoid generating bad code. Is this bound mandatory in the trait definition in question here? Why is it not required later when implementing the trait ?

Reply:
Rust complier can infer this is Sized because you're implementing the trait for a concert Sized struct. If you are using that in another trait, then the compiler cannot make assumptions and need to provide the bound.

Reply:
okay thanks

Reply:
i'll remove them